src/NServiceBus.Core/Routing/FileBasedDynamicRouting/FileRoutingTable.cs

```diff
@@ -98,13 +99,11 @@ protected override Task OnStop(IMessageSession session)
         string filePath;
         string errorMessage;
 
-        Dictionary<string, Task<IEnumerable<EndpointInstance>>> instanceMap = new Dictionary<string, Task<IEnumerable<EndpointInstance>>>();
+        Dictionary<string, IEnumerable<EndpointInstance>> instanceMap;
```

Comment:
This could be `Dictionary<string, HashSet<EndpointInstance>>` and then you wouldn't need the cast in `ReloadData`.